semantic-core/src/Analysis/Concrete.hs

```diff
@@ -20,7 +20,6 @@ import           Control.Effect.NonDet
 import           Control.Effect.Reader hiding (Local)
 import           Control.Effect.State
 import           Control.Monad ((<=<), guard)
-import qualified Data.Core as Core
```

Comment:
We no longer rely directly on imports of `Data.Core` or any other syntax. We _do_ still rely on `Analysis.Eval` for the `Analysis` record, but it could easily be moved elsewhere (and will eventually be factored into effects anyway).

This means that we could in principle split the analyses off into a new package, which is a good sign that we’ve got the right separation of concerns going.